### PR TITLE
[Core] Defining `GetBoundingBox` in `Tree` class. Plus adding documentation

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

This PR makes several changes to the Tree class in kratos/spatial_containers/tree.h to enhance its functionality and clarity. It includes:

- Defining `GetBoundingBox` method in `Tree` class
- Defining `SizeType`, `IndexType`, and `CoordinateType`, ... as aliases with `using`.
- Adding documentation for member functions: `ExistPoint`, `SearchNearestPoint`, `SearchInRadius`, and `SearchInBox`.
- Updating comments and documentation for better code understanding in general.
- Minor code formatting improvements.

Also extending `BoundingBox` class documentation.

**🆕 Changelog**

- [Defining GetBoundingBox in Tree class. Plus adding documentation](https://github.com/KratosMultiphysics/Kratos/commit/0dad11439ee0dccb583886a7739e791681f78405)
- [Extend doc BB class](https://github.com/KratosMultiphysics/Kratos/pull/11730/commits/e6de76f4eb387afbda9d0e3492a5f7ba1d927a6b)

### CODE REVIEW

**Comment by loumalouomega** [comment]
@roigcarlo ?

File: kratos/spatial_containers/tree.h [hunk 1]
```diff
@@ -396,16 +474,35 @@ class Tree
...
     {
         return mBoundingBoxHighPoint;
     }
 
+    /**
+     * @brief Get the bounding box.
+     * @details This function creates a bounding box using the low and high points and returns it.
+     * @return The bounding box.
+     */
+    BoundingBox<PointType> GetBoundingBox()
```
Comment (roigcarlo) [inline kratos/spatial_containers/tree.h]: Better to call it `CalculateBoundingBox`, as its not getting the result from the class, its being calculated on every call. (or store it and then return `mBoundingBox`)

Comment (loumalouomega) [reply kratos/spatial_containers/tree.h]: Better the second option, wait

Comment (loumalouomega) [reply kratos/spatial_containers/tree.h]: Done

**Review by roigcarlo — APPROVED** [review]

**Pull request merged**